csharp/extractor/Semmle.Extraction.CSharp/Entities/Expression.cs

```diff
@@ -34,6 +34,11 @@ internal Expression(IExpressionInfo info, bool shouldPopulate = true)
 
         protected sealed override void Populate(TextWriter trapFile)
         {
+            if (Context.OnlyScaffold)
```

Comment:
Is this meant as a safe-guard, in case there is a place where we create an expression entity that we shouldn't have? In those cases one will then have a dangling reference, but perhaps that's better than having duplicate expressions.